modbuscommander/integrationpluginmodbuscommander.cpp

```diff
@@ -387,7 +387,15 @@ void IntegrationPluginModbusCommander::thingRemoved(Thing *thing)
     qCDebug(dcModbusCommander()) << "Removing thing" << thing->name();
     if (thing->thingClassId() == modbusTCPClientThingClassId) {
         ModbusTCPMaster *modbus = m_modbusTCPMasters.take(thing);
+        if (modbus) {
+            modbus->deleteLater();
+        }
+    } else if (thing->thingClassId() == modbusRTUClientThingClassId) {
+        ModbusRtuMaster *modbus = m_modbusRtuMasters.take(thing);
         modbus->deleteLater();
+        if (modbus) {
+            modbus->deleteLater();
```

Comment:
this doesn't make a lot of sense without removing line 395. But I still wonder why/how this could possibly happen. Seems like the more correct fix would make sure that a successful setupThing() will *always* have the modbus object in the list.